### PR TITLE
Feat: decodeContent in Request.apply

### PR DESCRIPTION
**Pull request by dinojohn**
Add Combine operator in HttpData[R,E]

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/experiment/ContentDecoder.scala [hunk 1]
```diff
@@ -1,15 +1,25 @@
 package zhttp.experiment
 
+import io.netty.buffer.{ByteBuf, ByteBufUtil}
+import zhttp.http.HTTP_CHARSET
 import zio.{Chunk, Queue, UIO, ZIO}
 
-sealed trait ContentDecoder[-R, +E, +B] { self => }
+sealed trait ContentDecoder[-R, +E, +B] { self =>
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/experiment/ContentDecoder.scala:7]: `ContentDecoder[R, E, A, B]` A specify type of input

File: zio-http/src/main/scala/zhttp/experiment/ContentDecoder.scala [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+  def getContent(content: ByteBuf): ZIO[R, E, Option[B]]
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/experiment/ContentDecoder.scala:8]: remove `getContent` in companion object of ContentDecoder use pattern matching

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 3]
```diff
@@ -173,13 +173,13 @@ case class HttpApp[-R, +E](asHttp: Http[R, E, Request, Response[R, E]]) { self =
               val nState = ad.decoderState
 
               unsafeRunZIO(for {
-                (publish, state) <- run(Chunk.fromArray(content.array()), nState, isLast)
-                _                <- publish match {
+                a <- run(Chunk.fromArray(content.array()), nState, isLast)
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala:176]: revert this

File: zio-http/src/main/scala/zhttp/http/Request.scala [hunk 4]
```diff
@@ -43,7 +44,12 @@ trait Request extends HeadersHelpers { self =>
 }
 
 object Request {
-  def apply(method: Method = Method.GET, url: URL = URL.root, headers: List[Header] = Nil): Request = {
+  def apply(
+    method: Method = Method.GET,
+    url: URL = URL.root,
+    headers: List[Header] = Nil,
+    content: ByteBuf = Unpooled.EMPTY_BUFFER,
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/Request.scala:51]: content: A,
use TypeClass and define implicits for Byte, Chunk[Byte], Zstream[R, E, Byte]

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

**Comment by dinojohn** [comment]
closing in favour of #381

**Pull request closed without merging**